### PR TITLE
[docs] Update "Profiling Components with Chrome Timeline"

### LINKED ISSUE #9601: [docs] Update "Profiling Components with Chrome Timeline"
Google Chrome 58 just dropped and there is a shiny new **Performance** tab instead of the old **Timeline** tab in devTools. You can read all about it [here](https://developers.google.com/web/updates/2017/03/devtools-release-notes), but there is really not much to read except for:

> ### _"The Timeline panel has been renamed to the Performance panel, to better reflect its purpose."_

As a result, the docs for `optimizing-performance.md` needs to be updated [here](https://github.com/facebook/react/blob/master/docs/docs/optimizing-performance.md) to reflect these new changes. 

I tested by profiling a component in our React app at work, seems to work the same so it's really just a matter of updating the docs. I will submit a PR with the update 👍

### PR DESCRIPTION
**Pull request by himynameisdave**
As of Chrome 58, the Timeline tab is now called the Performance tab, this updates the "Optimizing Performance > Profiling Components with Chrome Performance" section of the docs to reflect that.

Resolves #9601

### CODE REVIEW

**Comment by himynameisdave** [comment]
@facebook-github-bot sorry I just signed it!

File: docs/docs/optimizing-performance.md [hunk 1]
```diff
@@ -160,17 +160,17 @@ You can learn more about this in [Webpack documentation](https://webpack.js.org/
 
 Remember that you only need to do this for production builds. You shouldn't apply `UglifyJsPlugin` or `DefinePlugin` with `'production'` value in development because they will hide useful React warnings, and make the builds much slower.
 
-## Profiling Components with Chrome Timeline
+## Profiling Components with Chrome Performance
```
Comment (nhunzaker) [inline docs/docs/optimizing-performance.md]: **Nitpick:**
"Profiling Components with Chrome Performance" reads funny to me. What about something like: "Profiling Components with the Chrome Performance Tab"

**Comment by nhunzaker** [comment]
@himynameisdave Do you mind resolving the conflicts with master?

File: docs/docs/optimizing-performance.md (hunk 1 above)
Comment (himynameisdave) [inline docs/docs/optimizing-performance.md]: Yep no problem, I'll fix that up then resolve & merge with master 👍

**Comment by himynameisdave** [comment]
@nhunzaker all cleaned up now 👍

**Comment by nhunzaker** [comment]
Looks great. Thank you for your contribution!

**Comment by gaearon** [comment]
@nhunzaker Please add "needs merge to stable" label to merged doc PRs, or changes won't appear until React 16.

**Comment by nhunzaker** [comment]
Ah. Thanks! Will do.

**Pull request merged**